pennylane/gradients/finite_difference.py

```diff
@@ -0,0 +1,258 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains functions for computing the finite-difference gradient
+of a quantum tape.
+"""
+# pylint: disable=protected-access,too-many-arguments
+import numpy as np
+from scipy.special import factorial
+
+import pennylane as qml
+
+
+def finite_diff_stencil(n, order, form):
+    r"""Generate the finite difference stencil (shift and coefficients)
+    for various derivatives, accuracy, and form.
+
+    Args:
+        n (int): Positive integer specifying the derivative. ``n=1``
+            corresponds to the first derivative.
+        order (int): Positive integer referring to the approximation order of the
+            returned stencil. E.g., ``order=1`` corresponds to a stencil
+            that returns a first-order approximation to the derivative.
+        form (str): one of ``"forward"``, ``"center"``, or ``"backward"``.

```

Comment:
Is form the kind of derivative rule? Maybe explain above?

Reply:
Yep; forward, backward, center, etc. I wasn't sure how to explain it, but maybe I will simply describe it in terms of the starting point of the shift.

Reply:
I've extended the description here!